controllers/httproute_controller.go

```diff
@@ -379,3 +409,45 @@ func isExternalDNSSupported(mgr ctrl.Manager) (bool, error) {
 	}
 	return false, nil
 }
+
+func (r *HTTPRouteReconciler) areBackendRefsIpFamiliesValid(ctx context.Context, httpRoute *gateway_api.HTTPRoute) bool {
+	rules := httpRoute.Spec.Rules
+
+	for _, rule := range rules {
+		backendRefs := rule.BackendRefs
+
+		for _, backendRef := range backendRefs {
+			// For now we skip checking service import
+			if *backendRef.Kind == "ServiceImport" {
+				continue
+			}
+
+			svc := &corev1.Service{}
+
+			glog.Infof("backendRef: %v", backendRef)
+
+			key := types.NamespacedName{
+				// Namespace: string(*backendRef.Namespace),
+				Name: string(backendRef.Name),
+			}
+
+			if backendRef.Namespace != nil {
+				key.Namespace = string(*backendRef.Namespace)
+			} else {
+				key.Namespace = httpRoute.Namespace
+			}
+
+			if err := r.client.Get(ctx, key, svc); err != nil {
+				// Ignore error since Service might not be created yet
```

Comment:
if service is not created we should requeue reconcile request, otherwise you assume here that route is valid, even if it's dual stack

Reply:
I think creation of the service should trigger reconcile request anyways. Looks like this function should only determine if the ipFamily is valid.